remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/SSEClient.java

```diff
@@ -23,54 +23,71 @@
 
 package de.unistuttgart.iaas.amyassist.amy.remotesr;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import java.util.concurrent.ExecutionException;
 
 import javax.ws.rs.sse.Sse;
 import javax.ws.rs.sse.SseEventSink;
-import java.util.concurrent.ExecutionException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * SSE client module object
  *
- * @author Benno Krauß
+ * @author Benno Krauß, Tim Neumann
  */
 class SSEClient {
 
-    private final Logger logger = LoggerFactory.getLogger(SSEClient.class);
+	private final Logger logger = LoggerFactory.getLogger(SSEClient.class);
 
-    private final SseEventSink sink;
-    private final Sse sse;
+	private final SseEventSink sink;
+	private final Sse sse;
 
-    SSEClient(SseEventSink sink, Sse sse) {
-        this.sink = sink;
-        this.sse = sse;
-    }
+	/**
+	 * Creates a new SSE Client
+	 * 
+	 * @param sink
+	 *            The sse event sink
+	 * @param sse
+	 *            The sse
+	 */
+	SSEClient(SseEventSink sink, Sse sse) {
+		this.sink = sink;
+		this.sse = sse;
+	}
 
-    boolean isConnected() {
-        return !sink.isClosed();
-    }
+	/**
+	 * @return Whether this client is connected.
+	 */
+	boolean isConnected() {
+		return this.sink != null && !this.sink.isClosed();
```

Comment:
`this.sink` is final, only set once in the constructor. Initializing an SSEClient with a null-EventSink doesn't make sense, so neither does this check. Instead, consider adding @NonNull to the constructor parameter